packages/react/src/components/Popover/index.test.tsx

```diff
@@ -0,0 +1,311 @@
+
+import type { PopoverProps } from '../Popover';
+
+let wrapperNode: HTMLDivElement | null;
+beforeEach(() => {
+  wrapperNode = document.createElement('div');
+  wrapperNode.innerHTML = `
+    <button data-test>Click Me!</button>
+    <div></div>
+  `;
+  document.body.appendChild(wrapperNode);
+});
+
+afterEach(() => {
+  document.body.innerHTML = '';
+  wrapperNode = null;
+});
+
+const Wrapper = ({
+  buttonProps = {},
+  tooltipProps = {}
+}: {
+  buttonProps?: React.HTMLAttributes<HTMLButtonElement>;
+  tooltipProps?: Omit<Partial<PopoverProps>, 'variant'>;
+}) => {
+  const ref = useRef<HTMLButtonElement>(null);
+  const onClose = jest.fn();
+
+  return (
+    <React.Fragment>
+      <button ref={ref} {...buttonProps}>
+        button
+      </button>
+      <Popover
+        variant="custom"
+        aria-labelledby="test-popover-title"
+        target={ref}
+        show
+        onClose={onClose}
+        {...tooltipProps}
+      >
+        Hello World
+      </Popover>
+    </React.Fragment>
+  );
+};
+
+const WrapperPopoverWithElements = () => {
+  const ref = useRef<HTMLButtonElement>(null);
+  const onClose = jest.fn();
+  return (
+    <React.Fragment>
+      <button ref={ref}>button</button>
+      <Popover
+        variant="custom"
+        aria-labelledby="test-popover-title"
+        target={ref}
+        show
+        onClose={onClose}
+      >
+        <button data-testid="foo1">Foo1</button>
+        <button data-testid="foo2">Foo2</button>
+        <button data-testid="foo3">Foo3</button>
+      </Popover>
+      <div data-testid></div>
+    </React.Fragment>
+  );
+};
+
+const WrapperPrompt = ({
+  buttonProps = {},
+  tooltipProps = {}
+}: {
+  buttonProps?: React.HTMLAttributes<HTMLButtonElement>;
+  tooltipProps?: Omit<Partial<PopoverProps>, 'variant'>;
+}) => {
+  const ref = useRef<HTMLButtonElement>(null);
+  const onClose = jest.fn();
+  const onApply = jest.fn();
+
+  return (
+    <React.Fragment>
+      <button ref={ref} {...buttonProps}>
+        button
+      </button>
+      <Popover
+        show
+        variant="prompt"
+        target={ref}
+        infoText="popover"
+        onApply={onApply}
+        onClose={onClose}
+        {...tooltipProps}
+      />
+    </React.Fragment>
+  );
+};
+
+test('renders without blowing up', async () => {
+  render(<Wrapper />);
+  expect(screen.getByText('Hello World')).toBeTruthy();
+});
+
+test('should auto-generate id', async () => {
+  render(<Wrapper />);
+  const popover = screen.getByText('Hello World').closest('.Popover');
+  const button = screen.getByText('button');
+  expect(popover).toBeTruthy();
+  const id = popover?.getAttribute('id');
+  expect(id).toBeTruthy();
+  expect(id).toEqual(button.getAttribute('aria-controls'));
+});
+
+test('should attach attribute aria-expanded correctly based on shown state', async () => {
+  const { rerender } = render(<Wrapper />);
+  const button = screen.getByText('button');
```

Comment:
This could be enhanced to be more explicit:

```ts
expect(screen.queryByRole('button', { name: 'button', expanded: true })).toBeInTheDocument()
rerender(<Wrapper tooltipProps={{ show: false }} />);
expect(screen.queryByRole('button', { name: 'button', expanded: false })).toBeInTheDocument()
```